src/wp-includes/block-template.php

```diff
@@ -129,6 +131,43 @@ static function ( $template_a, $template_b ) use ( $slug_priorities ) {
 		}
 	);
 
+	$theme_base_path        = get_stylesheet_directory() . DIRECTORY_SEPARATOR;
+	$parent_theme_base_path = get_template_directory() . DIRECTORY_SEPARATOR;
+
+	// Is the current theme a child theme, and is the PHP fallback template part of it?
+	if (
+		strpos( $fallback_template, $theme_base_path ) === 0 &&
+		strpos( $fallback_template, $parent_theme_base_path ) === false
+	) {
+		$fallback_template_slug = substr(
+			$fallback_template,
+			// Starting position of slug.
+			strpos( $fallback_template, $theme_base_path ) + strlen( $theme_base_path ),
+			// Remove '.php' suffix.
+			-4
+		);
+
+		// Is our candidate block template's slug identical to our PHP fallback template's?
+		if (
+			count( $templates ) &&
+			$fallback_template_slug === $templates[0]->slug &&
```

Comment:
Is it possible that the fallback is actually different than the block template slug but has more priority. Ex: `single.php`, `index.html`. Which one would win here?

Reply:
> Is it possible that the fallback is actually different than the block template slug but has more priority.

No, we prevent that by truncating the list of candidate templates that we pass from `locate_block_template()` to `resolve_block_template()` to only include equal or higher specificity than the PHP fallback template:

https://github.com/WordPress/wordpress-develop/blob/8bd1509ea8d1120b72256929c57bab6701d7ee58/src/wp-includes/block-template.php#L25-L48

Reply:
Ok I guess this looks good to me then.